zerver/models.py

```diff
@@ -184,6 +184,9 @@ class Realm(models.Model):
     name_changes_disabled = models.BooleanField(default=False)  # type: bool
     email_changes_disabled = models.BooleanField(default=False)  # type: bool
 
+    # Whether the delivery_email is hidden from the UI
+    delivery_email_hidden = models.BooleanField(default=False)  # type: bool
```

Comment:
I think we could have a clearer name for this, e.g. `email_addresses_are_public` with `(default=True)` or something.

@rishig do you have thoughts on naming this?

Reply:
`email_addresses_are_public` seems reasonable. I guess this could also be `email_address_visibility` if we imagine we'll add more possible settings, though I can't think of any right now. (Presumably for the web-public view we'll always hide email addresses.)

Reply:
I guess I'm imagining the setting for this could be

`Who can see other users' email addresses?`
* Admins, members, and guests
* Admins and members
* Admins only

Reply:
OK, seems reasonable.  So we should set this up like `bot_creation_policy`, with values like:
```
EMAIL_ADDRESS_VISIBILITY_EVERYONE
EMAIL_ADDRESS_VISIBILITY_MEMBERS
EMAIL_ADDRESS_VISIBILITY_ADMINS
```